packages/back-end/src/api/openapi/payload-schemas/PostDimensionPayload.yaml

```diff
@@ -0,0 +1,32 @@
+type: object
+required:
+  - datasourceId
+  - identifierType
+  - name
+  - query
+properties:
+  name:
+    type: string
+    description: Name of the dimension
+  description:
+    type: string
+    description: Description of the dimension
+  owner:
+    type: string
+    description: Owner of the dimension
+  datasourceId:
+    type: string
+    description: ID of the datasource this dimension belongs to
+  identifierType:
+    type: string
+    description: Type of identifier (user, anonymous, etc.)
+  query:
+    type: string
+    description: SQL query or equivalent for the dimension
+  managedBy:
+    type: string
+    description: Where this dimension must be managed from. If not set (empty string), it can be managed from anywhere.
+    enum:
+      - ""
+      - api
+      - config
```

Comment:
I think we may want to remove the "config" option from the enum as I don't think we want to encourage users to create/update dimensions via the REST api and set `managedBy: "config"` as they'd get in a really weird state where it couldn't be removed/edited.

I think removing the config option here might make some of the linters unhappy, but I think we solved this issue with the Metrics endpoints -

Reply:
good call. This also has the side effect of updating the zod validator, giving us endpoint safety (you can't pass in "config" to either create or update endpoints)